### ISSUE TITLE
Support Multi-Cluster Ingress

### ISSUE THREAD

**Issue by sacrex** [karmada-io/karmada#1079]
**Please provide an in-depth description of the question you have**:
   K8s has signal-cluster ingress controller that forward traffic to services. 
   Does Karmada have mulit-cluster counterparts or not?
**What do you think about this question?**:
   1.If necessary, how to integrate with karmada.
**Environment**:
- Karmada version: v0.10.1
- Kubernetes version: v1.22.3
- Others:

**Comment by GitHubxsy** [comment]
On  roadmap. kubernetes-sigs defined Multi-cluster Service APIs（[MCS](https://github.com/kubernetes-sigs/mcs-api)）. However, the Mutil-cluster ingress(MCI) API is not provided. Do you have a good idea ? 
Do you have a scenario share?
My idea:
![image](https://user-images.githubusercontent.com/11901080/145036500-a0148387-20d9-4994-96ec-467a7a18ed97.png)

**Comment by sacrex** [comment]
1. Do you have a good idea ?
Your idea is very good, and i think so almost. However, I have a little question that how the MultiClusterService in Karmada interact with the Server in member cluster which is deployed on the private cloud?
2. Do you have a scenario share?
scenario: Now, the Karmada has taken over a cluster which deployed nginx webserver. If Karmada dynamically taken over another new cluster which deployed nginx in it (assume the origin cluster can't scale), how can the client's requests reach the new cluster? So I think multi-cluser LB is needed and use ingress to complete it.

**Comment by GitHubxsy** [comment]
> 1. Do you have a good idea ?
>    Your idea is very good, and i think so almost. However, I have a little question that how the MultiClusterService in Karmada interact with the Server in member cluster which is deployed on the private cloud?
> 2. Do you have a scenario share?
>    scenario: Now, the Karmada has taken over a cluster which deployed nginx webserver. If Karmada dynamically taken over another new cluster which deployed nginx in it (assume the origin cluster can't scale), how can the client's requests reach the new cluster? So I think multi-cluser LB is needed and use ingress to complete it.

MultiClusterService requires that the cluster container network is Intercommunication. For example, using [submariner](https://github.com/submariner-io/submariner)to open up the cluster network.

**Comment by GitHubxsy** [comment]
@sacrex  We will discuss MultiClusterIngress at the community meeting on January 18. You are welcome to join the [meeting](https://docs.google.com/document/d/1y6YLVC-v7cmVAdbjedoyR5WL0-q45DBRXTvz5_I7bkA/edit#).

**Comment by GitHubxsy** [comment]
![image](https://user-images.githubusercontent.com/11901080/149899933-9c47f7d0-72e2-49d4-947a-eca1e2197614.png)

MCI API
```
apiVersion: work.karmada.io/v1alpha1
kind: MultiClusterIngress
metadata:
  name: mci-demo
spec:
  rules:
  - http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: test  # MultiClusterService name
status:
  loadBalancer:
    ingress:
    - ip: <LBVIP>
```

**Comment by sacrex** [comment]
> @sacrex We will discuss MultiClusterIngress at the community meeting on January 18. You are welcome to join the [meeting](https://docs.google.com/document/d/1y6YLVC-v7cmVAdbjedoyR5WL0-q45DBRXTvz5_I7bkA/edit#).

I'm sorry that I didn't attend the meeting last week.
Thank you very much for your solution.
By the way, when can use it? Thanks again~

**Comment by GitHubxsy** [comment]
> > @sacrex We will discuss MultiClusterIngress at the community meeting on January 18. You are welcome to join the [meeting](https://docs.google.com/document/d/1y6YLVC-v7cmVAdbjedoyR5WL0-q45DBRXTvz5_I7bkA/edit#).
> 
> I'm sorry that I didn't attend the meeting last week. Thank you very much for your solution. By the way, when can use it? Thanks again~

The next release.

**Comment by RainbowMango** [comment]
It will be included in `v1.1` release(by the end of Feb 2022).

**Comment by XiShanYongYe-Chang** [comment]
Reopen it to record the feature development process.
